example_code/shmem_put_signal_example.c

```diff
@@ -0,0 +1,42 @@
+#include <stdio.h>
+#include <shmem.h>
+#include <stdlib.h>
+
+int main(void)
+{
+    int i, err_count  = 0;
+
+    shmem_init();
+
+    size_t    size    = (2<<10);
+    int       me      = shmem_my_pe();
+    int       n       = shmem_n_pes();
+    int       pe      = (me + 1)%n;
+    uint64_t* message = malloc(size * sizeof(uint64_t));
+    uint64_t* data    = shmem_malloc(size * sizeof(uint64_t));
```

Comment:
Why not use `shmem_calloc`?  You can remove `data[i] = 0` as well as `shmem_barrier_all()` below.

Reply:
I would move the declaration of `sig_addr` above the `shmem_calloc` if you make this change.  I can't convince myself that isn't a race without consulting the C language spec.

Reply:
resolved - made changes as per the suggestion. 
closing outdated changes.